src/algokit/cli/init.py

```diff
@@ -216,6 +217,15 @@ def init_command(  # noqa: PLR0913
     # provide the directory name as an answer to the template, if not explicitly overridden by user
     answers_dict.setdefault("project_name", directory_name)
 
+    system_python_path = next(get_python_paths(), None)
+    if system_python_path is not None:
```

Comment:
@achidlow @negar-abbasi - what if we expand this and also add usage of `is_binary_mode` and if its TRUE we execute a python script using something like [this](https://github.com/pyinstaller/pyinstaller/issues/2490) via `exec` i believe it should refer to currently used interpreter which in case of pyinstaller will be the one bundled along with the bootloader. That way users of binary installed algokit won't need to have python to run custom tasks in copier

Reply:
I think the solution in this PR is simpler. Adding another alogkit command which is called from a copier template `_tasks` to execute a Python script that is part of template's data is a very tight coupling compared to informing a template of a Python interpreter location on the `$PATH` via a supplied answer.

Reply:
@achidlow i see so this is all handled by `get_python_paths`, if `which` did not return `python` or `python3` it will grab the `sys.executable` which will point to the interpreter bundled in binary (when running in binary mode).

Reply:
ah, we need to update that to return `None` if `sys.executable and sys.executable.endswith("alogkit")` or something like that, it's not a valid interpreter so it shouldn't be returned. This goes for templates here as well as where it's used for `pipx`